### PR TITLE
[0.9] Removing `instrumentation_keys` in order to fix the payload

### PR DESCRIPTION
**Pull request by yosiat**
Fixes - https://github.com/rails-api/active_model_serializers/issues/2067

By the way, if the instrumentation is always the same - `!serialize.active_model_serializers`,
I think a nice change would be to create a const for `INSTRUMENTATION_KEY` that will have `!serialize.active_model_serializers` as a frozen string value.
This improves performance & reduces allocations by a ton.

I can do that if the author wants that.

### CODE REVIEW

**Comment by bf4** [comment]
Looks fine to me. However, any thought of using `render.active_model_serializers` instead of `!serialize.active_model_serializers`? As you said, it's currently broken in 0.9.

Would you mind adding tests and yourself in the changelog?

cc @wagenet ref https://github.com/skylightio/skylight-ruby/blob/v1.1.0/lib/skylight/probes/active_model_serializers.rb#L34-L47

```ruby
          # End users could override as_json without calling super, but it's likely safer
          # than overriding serializable_array/hash/object.

          [::ActiveModel::Serializer, ::ActiveModel::ArraySerializer].each do |klass|
            klass.class_eval do
              alias as_json_without_sk as_json
              def as_json(*args)
                payload = { serializer: self.class }
                ActiveSupport::Notifications.instrument('render.active_model_serializers', payload) do
                  as_json_without_sk(*args)
                end
              end
            end
          end
```

**Comment by yosiat** [comment]
@bf4 The instrumentation today works, you only get the `serializer` in the payload.
so changing the name is a breaking behaviour.

**Comment by bf4** [comment]
Would you mind adding tests and yourself in the changelog?

**Comment by yosiat** [comment]
@bf4 yes, I will add tests at the weekend.

**Comment by yosiat** [comment]
@bf4 I saw that today we use the instrumentation in assert_serializer.
So If I broke somehow the instrumentation than assert_serializer will get broken, so I don't think there is a need for tests here.
If you still me want me to add some sanity tests, I will add.

By the way, here is the status of the benchmarks:

**0-9-stable:**
```
Single: DefaultSerializer 27187/ips; 76 objects
ArraySerializer 14801/ips; 130 objects
ArraySerializer: each_serializer: DefaultSerializer 18183/ips; 122 objects
FlatAuthorSerializer 27428/ips; 85 objects
ArraySerializer: each_serializer: FlatAuthorSerializer 19851/ips; 114 objects
AuthorWithDefaultRelationshipsSerializer 320/ips; 6190 objects
ArraySerializer: each_serializer: AuthorWithDefaultRelationshipsSerializer 331 ip/s; 6219 objects
```

**this branch:**
```
Single: DefaultSerializer          32682 ip/s         67 allocs/op
ArraySerializer    20902 ip/s         89 allocs/op
ArraySerializer: each_serializer: DefaultSerializer        24746 ip/s         81 allocs/op
FlatAuthorSerializer       36517 ip/s         62 allocs/op
ArraySerializer: each_serializer: FlatAuthorSerializer     29672 ip/s         68 allocs/op
AuthorWithDefaultRelationshipsSerializer             397 ip/s       5258 allocs/op
ArraySerializer: each_serializer: AuthorWithDefaultRelationshipsSerializer           397 ip/s       5264 allocs/op
```

**Comment by bf4** [comment]
@yosiat Yes, I'd love tests!

File: test/unit/active_model/serilizable_test.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+module ActiveModel
+  class SerializableTest
+    class InstrumentationTest < Minitest::Test
+      def setup
+        @events = []
+
+        ActiveSupport::Notifications.subscribe("!serialize.active_model_serializers") do |name, start, finish, id, payload|
```
Comment (bf4) [inline test/unit/active_model/serilizable_test.rb]: Better:

`@subscriber = ActiveSupport::Notifications.subscribe(.....` in the setup

and

`ActiveSupport::Notifications.unsubscribe(@subscriber) if defined?(@subscriber)` in the teardown.  less global.

File: test/unit/active_model/serilizable_test.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      end
+
+      def teardown
+        ActiveSupport::Notifications.unsubscribe("!serialize.active_model_serializers")
+      end
+
+      def test_default_serializer
+        DefaultSerializer.new(1).as_json
+
+        assert_equal @events, [{ name: '!serialize.active_model_serializers', serializer: 'ActiveModel::DefaultSerializer' }]
```
Comment (bf4) [inline test/unit/active_model/serilizable_test.rb]: expectation in assert_equal should be on the left http://docs.seattlerb.org/minitest/Minitest/Assertions.html

File: test/unit/active_model/serilizable_test.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        ActiveSupport::Notifications.subscribe("!serialize.active_model_serializers") do |name, start, finish, id, payload|
+          @events << { name: name, serializer: payload[:serializer] }
+        end
+      end
+
+      def teardown
+        ActiveSupport::Notifications.unsubscribe("!serialize.active_model_serializers")
+      end
+
+      def test_default_serializer
```
Comment (bf4) [inline test/unit/active_model/serilizable_test.rb]: maybe `test_instruments_default_serializer` here and below, I think would better description of the test

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 ### [0-9-stable](https://github.com/rails-api/active_model_serializers/compare/v0.9.5...0-9-stable)
 
+- [#2080](https://github.com/rails-api/active_model_serializers/pull/2080) remove `{ payload: nil }` from `!serialize.active_model_serializers` ActiveSupport::Notification. `payload` never had a value.  Changes, for example `{ serializer: 'ActiveModel::DefaultSerializer', payload: nil }` to be `{ serializer: 'ActiveModel::DefaultSerializer' }` (@yosiat)
```
Comment (bf4) [inline CHANGELOG.md]: 👍

**Pull request merged**